packages/playwright-core/src/cli/cli.ts

```diff
@@ -309,13 +309,28 @@ Examples:
 
 if (!process.env.PW_LANG_NAME) {
   let playwrightTestPackagePath = null;
+  const resolvePwTestPaths = [__dirname, process.cwd()];
   try {
     playwrightTestPackagePath = require.resolve('@playwright/test/lib/cli', {
-      paths: [__dirname, process.cwd()]
+      paths: resolvePwTestPaths,
     });
   } catch {}
 
   if (playwrightTestPackagePath) {
+    const pwTestVersion = require(require.resolve('@playwright/test/package.json', {
+      paths: resolvePwTestPaths,
+    })).version;
+    const pwCoreVersion = require(path.join(__dirname, '../../package.json')).version;
+    if (pwTestVersion !== pwCoreVersion) {
+      console.error(wrapInASCIIBox([
+        `Playwright Test compatibility check failed:`,
+        `@playwright/test version '${pwTestVersion}' does not match playwright-core version '${pwCoreVersion}'!`,
+        `To fix this either align the Playwright versions or only keep @playwright/test since it depends on playwright-core.`,
+        `If you still receive this error, execute 'npm ci' or delete 'node_modules' and do 'npm install' again.`,
+      ].join('\n')));
```

Comment:
What are some scenarios the user ends up here? Like how do they end up with mismatched versions in the first place. It happens, and I'd like to understand why since maybe we can catch it on the other end, too.

Reply:
Most common scenario is if they have pw-core 1.24 and pwt 1.25

Maybe also playwright the package 1.24 and pwt 1.25.

Reply:
That's the issue, but it doesn't explain how they end up with pw-core 1.24 and pwt 1.25. Like what sequence of command would have had to be run to get into this situation?

Reply:
> That's the issue, but it doesn't explain how they end up with pw-core 1.24 and pwt 1.25. Like what sequence of command would have had to be run to get into this situation?

In both my my examples I imply that they have these in the package.json.